### PR TITLE
fix(billing): adapt billing gauge for 3000

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

We broke the billing gauges with 3000. This PR aims to fix the gauges and give them a 3000 refresh.

See also https://posthog.slack.com/archives/C064U4XCPHQ/p1701799190530919 and since this is a private thread:
Screenshot 2023-12-08 at 10 59 06

## Changes

This is how the current billing gauge is built:

Screenshot 2023-12-08 at 10 56 19

This is how the 3000 billing gauge is built:
Screenshot 2023-12-08 at 11 14 29

## Screenshots

Resulting in the following gauges (**Note that the visual regression tests for the 3000 variant don't work, as the storybook way of configuring the feature flag isn't working for multi-variate feature flags, only boolean ones**):

**Legacy** - same as before.
Screenshot 2023-12-08 at 16 22 57

**Light mode, within billing limit**
Screenshot 2023-12-08 at 16 22 24
**Light mode, over billing limit** (ignore the tick on the bar, I hackily forced the state)
Screenshot 2023-12-08 at 16 22 11
**Dark mode, within billing limit**
Screenshot 2023-12-08 at 16 21 12
**Dark mode, over billing limit**
Screenshot 2023-12-08 at 16 21 57

## How did you test this code?

Mostly tested with storybooks - let me know if there are edge cases that I should check

### CODE REVIEW

File: frontend/src/scenes/billing/BillingGauge.scss [hunk 1]
```diff
@@ -23,6 +23,55 @@
...
+        background: var(--danger);
+
+        &.BillingGaugeItem--within-usage-limit {
+            background: var(--brand-blue);
+        }
+    }
+
+    &.BillingGaugeItem--projected_usage {
+        background: repeating-linear-gradient(
+            -45deg,
```
Comment (thmsobrmlr) [inline frontend/src/scenes/billing/BillingGauge.scss:57]: Tried to mimic the same stripe pattern we have for funnels / area charts, but changed the angle. Judge yourself:

**Adjusted**
Screenshot 2023-12-08 at 16 21 12


**Funnels/Area Charts**
Screenshot 2023-12-08 at 16 31 26

**Review by corywatilo — APPROVED** [review]
Much better, thank you!

File: frontend/src/scenes/billing/BillingGauge.scss [hunk 2]
```diff
@@ -23,6 +23,55 @@
...
+
+    &.BillingGaugeItem--free_tier {
+        // :TRICKY: We want "--border-light" as background color here.
+        // Since this color has an alpha-component, it "stacks" with
+        // other colors. Therefore we're using the effective color
+        // without alpha component as replacement.
+        background-color: #dedede;
+
+        [theme='dark'] & {
+            background-color: #1b1d24;
```
Comment (raquelmsmith) [inline frontend/src/scenes/billing/BillingGauge.scss:35]: Should we use a var here instead? Eg. `--danger-opacity-1`

**Review by raquelmsmith — COMMENTED** [review]
Thank you for doing this! Code looks fine, I'll trust you that it works. 

However, I feel like the blue and red colors are really hard to see in dark mode. can we use lighter color to give more contrast?

**Comment by thmsobrmlr** [comment]
> However, I feel like the blue and red colors are really hard to see in dark mode. can we use lighter color to give more contrast?

I kind of agree, though it's not super straight forward to fix. Making the color lighter automatically makes it more white. An alternative would be to make the background darker, and given that we have similar comments in other places this might be more appropriate. I'll come up with a separate PR, to not blow up the scope here.

**Pull request merged**